app/src/androidTest/java/com/github/wnder/GetUserPositionTest.java

```diff
@@ -106,5 +106,7 @@ public void PositionIsWithinRange(){
 
         assertThat(l1.getLatitude(), Matchers.is(50.0));
         assertThat(l2.getLatitude(), Matchers.is(60.0));
+
+        locationManager.removeTestProvider(LocationManager.GPS_PROVIDER);
```

Comment:
Maybe you should put it in a function with an @After to be sure that it is really removed even if the test crash.